votor/src/certificate_pool/slot_stake_counters.rs

```diff
@@ -0,0 +1,301 @@
+use {
+    crate::{
+        certificate_pool::stats::CertificatePoolStats, event::VotorEvent, Stake,
+        SAFE_TO_NOTAR_MIN_NOTARIZE_AND_SKIP, SAFE_TO_NOTAR_MIN_NOTARIZE_FOR_NOTARIZE_OR_SKIP,
+        SAFE_TO_NOTAR_MIN_NOTARIZE_ONLY, SAFE_TO_SKIP_THRESHOLD,
+    },
+    solana_sdk::hash::Hash,
+    solana_votor_messages::vote::Vote,
+    std::collections::BTreeMap,
+};
+
+#[derive(Debug, Default)]
+pub(crate) struct SlotStakeCounters {
+    my_first_vote: Option<Vote>,
+    total_stake: Stake,
+    skip_total: Stake,
+    notarize_total: Stake,
+    notarize_entry_total: BTreeMap<Hash, Stake>,
+    top_notarized_stake: Stake,
+    safe_to_notar_sent: Vec<Hash>,
+    safe_to_skip_sent: bool,
+}
+
+impl SlotStakeCounters {
+    pub fn add_vote(
+        &mut self,
+        vote: &Vote,
+        total_stake: Stake,
+        entry_stake: Stake,
+        is_my_own_vote: bool,
+        events: &mut Vec<VotorEvent>,
+        stats: &mut CertificatePoolStats,
+    ) {
+        match vote {
+            Vote::Skip(_) => self.skip_total = entry_stake,
+            Vote::Notarize(vote) => {
+                self.notarize_entry_total
+                    .insert(*vote.block_id(), entry_stake);
+                self.notarize_total = self.notarize_entry_total.values().sum();
+                self.top_notarized_stake = self.top_notarized_stake.max(entry_stake);
+            }
+            _ => return, // Not interested in other vote types
```

Comment:
I think we can separate this into two separate functions 

`add_skip` and `add_notarize` that call a common method which encapsulates the logic below so that the argument list isn't as big